vms/rpcchainvm/vm_client.go

```diff
@@ -541,14 +541,20 @@ func (vm *VMClient) CrossChainAppRequest(ctx context.Context, chainID ids.ID, re
 	return err
 }
 
-func (vm *VMClient) CrossChainAppRequestFailed(ctx context.Context, chainID ids.ID, requestID uint32) error {
-	_, err := vm.client.CrossChainAppRequestFailed(
-		ctx,
-		&vmpb.CrossChainAppRequestFailedMsg{
-			ChainId:   chainID[:],
-			RequestId: requestID,
-		},
-	)
+func (vm *VMClient) CrossChainAppRequestFailed(ctx context.Context, chainID ids.ID, requestID uint32, err error) error {
+	msg := &vmpb.CrossChainAppRequestFailedMsg{
+		ChainId:      chainID[:],
+		RequestId:    requestID,
+		ErrorCode:    common.ErrUnexpected.Code,
+		ErrorMessage: common.ErrUnexpected.Message,
+	}
+
+	if appErr, ok := err.(*common.AppError); ok {
+		msg.ErrorCode = appErr.Code
+		msg.ErrorMessage = appErr.Message
+	}
```

Comment:
```suggestion
	if appErr, ok := err.(*common.AppError); ok {
	        // Provide the specific error instead of generic [common.ErrUnexpected]
		msg.ErrorCode = appErr.Code
		msg.ErrorMessage = appErr.Message
	}
```